design/ceph/admission-controller.md

```diff
@@ -0,0 +1,93 @@
+#Admission Controllers on Rook
+
+#Background
+This proposal is to add support for admission controllers in Rook. 
+
+Currently, user can manipulate CRD specs with any values which may result in application not functioning as expected. An admission controller is a piece of code that intercepts requests to the Kubernetes API server prior to persistence of the object, but after the request is authenticated and authorized. In that list, there are two special controllers: MutatingAdmissionWebhook and ValidatingAdmissionWebhook. These execute the mutating and validating (respectively) admission control webhooks which are configured in the API.
+
+The admission control process proceeds in two phases. In the first phase, mutating admission controllers are run. In the second phase, validating admission controllers are run. Note again that some of the controllers are both.
+
+If any of the controllers in either phase reject the request, the entire request is rejected immediately and an error is returned to the end-user.
+
+# Requirements
+
+1. Webhook server which will validate the requests
+2. TLS certificates for the server,
+3. ValidatingWebhookConfig/MutatingWebhookConfig which will intercept requests and send a HTTPS request to the webhook server.
+4. RBAC Components 
+
+#Admission Controllers in Rook
+
+![AC Image](./design-pic.png)
+
+##Setup
+###Certificate Management
+####Development and Testing 
+![AC Image](./ac_dev_design.png)
+
+We will be opting for using for self-signed certificates approved by the kubernetes cluster itself. Once common.yaml and the operator has been deployed.  Using scripts we will be performing the following tasks
+* Creating the private key and certs using openssl with the names tls.key and tls.crt. We will be hardcoding these names for now as it will be explained in later section
```

Comment:
The filenames are just for working with the script, right?

Reply:
Actually in the server code, we are looking for these files specifically to complete the configuration hence if these names are changed, the server would throw an error.